src/stores/settings.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @copyright Copyright (c) 2020 Joas Schilling <coding@schilljs.com>
+ *
+ * @author Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { defineStore } from 'pinia'
+
+import { getCapabilities } from '@nextcloud/capabilities'
+import { loadState } from '@nextcloud/initial-state'
+
+import { PRIVACY } from '../constants.js'
+import { setReadStatusPrivacy, setTypingStatusPrivacy } from '../services/settingsService.js'
+
+export const useSettingsStore = defineStore('settings', {
+	state: () => ({
+		readStatusPrivacy: loadState('spreed', 'read_status_privacy', PRIVACY.PRIVATE),
+		typingStatusPrivacy: loadState('spreed', 'typing_privacy', PRIVACY.PRIVATE),
+		supportTypingStatus: getCapabilities()?.spreed?.config?.chat?.['typing-privacy'] !== undefined,
+	}),
```

Comment:
I didn't notice it at first.

`supportTypingStatus` is not an application state, it never changes, and we do not support capabilities updates without reloading.

I prefer not to store constants in a global reactive store.

Reply:
I wanted to ease access to constants from capabilities.
Can we do it another way, maybe? get them from server once, store as simple JS constant in, say, `src/capabilities.js`, and import from there?

Reply:
> Can we do it another way, maybe? get them from server once, store as simple JS constant in, say, `src/capabilities.js`, and import from there?

We can. But it seems, it won't change much.

If this module looks like

```js
export const talkCapabilities = {
  config: {
    chat: {
      typingPrivacy: getCapabilities()?.spreed?.config?.chat?.['typing-privacy'] !== undefined
    }
  }
}
```

then it will help to simplify one complex line

```js
getCapabilities()?.spreed?.config?.chat?.['typing-privacy'] !== undefined
```

to two simple

```js
import { talkCapabilities } from '...'

talkCapabilities.config.chat.typingPrivacy
```

Reply:
I personally would prefer to see "more original" capability path instead of new names.

```js
// Most explicit usage of capabilities
// But many optional chaining, long due to accessing spreed from all capabilities and casting to boolean by expression
getCapabilities()?.spreed?.config?.chat?.['typing-privacy'] !== undefined

// Close to capabilities
talkCapabilities.config.chat.typingPrivacy

// New name, required to jump into declaration
supportTypingPrivacy
```

We can discuss adding capabilities module. But speaking about this PR, I'm against the solution of mixing individual constant capabilities with global reactive state storages.

Reply:
Then I'll revert these changes for current PR for now

Reply:
Also it's not "support" typing privacy, but whether it's on/off for the current user?

Reply:
right, we have `spreed.feature.typing-privacy` as well as `spreed.config.chat.typing-privacy`. but yeah the config one is the user setting.

I'd say simplicity is key and we can keep it directly in the code, until we faced too many typos of one and then we do it with a method :D